packages/cactus-plugin-htlc-eth-besu-erc20/src/main/typescript/web-services/get-single-status-endpoint.ts

```diff
@@ -0,0 +1,88 @@
+import { Express, Request, Response } from "express";
+import {
+  Checks,
+  Logger,
+  LoggerProvider,
+  LogLevelDesc,
+} from "@hyperledger/cactus-common";
+import {
+  IExpressRequestHandler,
+  IWebServiceEndpoint,
+} from "@hyperledger/cactus-core-api";
+import { registerWebServiceEndpoint } from "@hyperledger/cactus-core";
+import {
+  PluginLedgerConnectorBesu,
+  Web3SigningCredentialType,
+  EthContractInvocationType,
+} from "@hyperledger/cactus-plugin-ledger-connector-besu";
+import HashTimeLockJSON from "../../solidity/contracts/HashedTimeLockContract.json";
+import { environment } from "../environment";
+
+export interface IGetSingleStatusEndpointOptions {
+  logLevel?: LogLevelDesc;
+  connector: PluginLedgerConnectorBesu;
+}
+
+export class GetSingleStatusEndpoint implements IWebServiceEndpoint {
+  public static readonly CLASS_NAME = "GetSingleStatusEndpoint";
+  private readonly log: Logger;
+  private readonly connector: PluginLedgerConnectorBesu;
+
+  constructor(public readonly options: IGetSingleStatusEndpointOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+    this.connector = this.options.connector;
+  }
+
+  public get className(): string {
+    return GetSingleStatusEndpoint.CLASS_NAME;
+  }
+
+  public getVerbLowerCase(): string {
+    return "get";
+  }
+
+  public getPath(): string {
+    return "/api/v1/plugins/@hyperledger/cactus-plugin-htlc-eth-besu-erc20/get-single-status/:id";
+  }
+
+  public registerExpress(expressApp: Express): IWebServiceEndpoint {
+    registerWebServiceEndpoint(expressApp, this);
+    return this;
+  }
+
+  public getExpressRequestHandler(): IExpressRequestHandler {
+    return this.handleRequest.bind(this);
+  }
+
+  public async handleRequest(req: Request, res: Response): Promise<void> {
+    const fnTag = "GetSingleStatusEndpoint#handleRequest()";
+    this.log.debug(`GET ${this.getPath()}`);
+    try {
+      const id = req.params["id"];
+      const result = await this.connector.invokeContract({
+        contractAbi: HashTimeLockJSON.abi,
+        contractAddress: environment.CONTRACT_ADDRESS,
+        invocationType: EthContractInvocationType.CALL,
+        methodName: "getSingleStatus",
+        params: [id],
+        web3SigningCredential: {
+          ethAccount: environment.ACCOUNT_ADDRESS,
+          secret: environment.PRIVATE_KEY,
+          type: Web3SigningCredentialType.PRIVATEKEYHEX,
+        },
+        gas: 6721975,
```

Comment:
@jordigiam Gas should also be customizable in case it's too much/not enough/etc.

Reply:
Sure, I will prepare the endpoints to expect the gas value, and I will leave this default value in case the user does not send the gas.

Also, I think the automated gas estimation maybe is a feat for the connector-besu, because in the htlc plugin we don't have an instance of the web3 with the address of the node, instead the connector is the one that has it.

https://web3js.readthedocs.io/en/v1.2.0/web3-eth.html#estimategas

We can discuss it

cc: @kikoncuo